pandas/tests/dtypes/test_common.py

```diff
@@ -731,3 +732,11 @@ def test__is_dtype_type_sparse():
     result = np.dtype("int32")
     assert com._is_dtype_type(ser, lambda tipo: tipo == result)
     assert com._is_dtype_type(ser.dtype, lambda tipo: tipo == result)
+
+
+def test__nansafe_nat_to_int():
+    arr = np.array([np.datetime64("NaT")])
```

Comment:
also paramaterize over np.timedelta('nat') and pd.NaT

Reply:
For this one I'm hitting some annoying snags.  `np.array([pd.NaT])` promotes `pd.NaT` to an `object`, and specifying the dtype with `np.array([pd.NaT], dtype="datetime64[ns]")` actually raises the error `ValueError: cannot convert float NaN to integer`.  The only way I've been able to get this to work is `pd.Series([pd.NaT]).values`, which is ugly (although the tests pass afterwards).  Any suggestions on the cleanest approach here?